zebra-chain/src/commitments/sapling.rs

```diff
@@ -82,7 +82,7 @@ impl NoteCommitment {
         diversifier: Diversifier,
```

Comment:
This seems odd to me. NoteCommit<sup>Sapling</sup> is a total function and takes g<sub>d</sub> as input, not d.

Reply:
It is but this method isn't quite that function as it requires an rng to be passed, the actual `NoteCommit^Sapling_rcm` is in the body, accepting the resulting rcm, the resulting g_d, the transmission_key  and the value, and then returns a tuple of the randomness and the note commitment. I can make that clear in the docstring